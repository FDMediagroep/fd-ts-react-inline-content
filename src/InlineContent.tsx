import {BulletPoints, BulletPointsStyle} from './components/BulletPoints';
import {LinkBlock, LinkBlockStyle} from './components/LinkBlock';
import {NumberFrame, NumberFrameStyle} from './components/NumberFrame';
import {Quote, QuoteStyle} from './components/Quote';
import {StackFrame, StackFrameStyle, StackFrameDataType} from './components/StackFrame';
import {Stock, StockStyle} from './components/Stock';
import {Summary, SummaryStyle} from './components/Summary';
import {TextFrame, TextFrameStyle} from './components/TextFrame';

export {BulletPoints, BulletPointsStyle};
export {LinkBlock, LinkBlockStyle};
export {NumberFrame, NumberFrameStyle};
export {Quote, QuoteStyle};
export {StackFrame, StackFrameStyle, StackFrameDataType};
export {Stock, StockStyle};
export {Summary, SummaryStyle};
export {TextFrame, TextFrameStyle};
