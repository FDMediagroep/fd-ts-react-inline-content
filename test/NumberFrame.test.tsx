import React from 'react';
import TestRenderer from 'react-test-renderer';
import { NumberFrame } from '../src/InlineContent';

describe('NumberFrame', ()  => {
    const title = "140%";
    const paragraph = "Curabitur quis luctus est, vitae sodales enim";

    it('should render align right correctly', () => {
        let component = TestRenderer.create(<NumberFrame {...{title, paragraph}} />);
        expect(component.toJSON()).toMatchSnapshot();

        component = TestRenderer.create(<NumberFrame {...{title, paragraph}} align='right'/>);
        expect(component.toJSON()).toMatchSnapshot();
    });

    it('should render align block correctly', () => {
        let component = TestRenderer.create(<NumberFrame {...{title, paragraph}} align='block'/>);
        expect(component.toJSON()).toMatchSnapshot();
    });

});
