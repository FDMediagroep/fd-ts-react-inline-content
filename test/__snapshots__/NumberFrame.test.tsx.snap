// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NumberFrame should render align block correctly 1`] = `
<div
  className="inline-content cyferkader block"
>
  <h3
    className="fd-typography article-h-inline-number-frame"
    textStyle="article-h-inline-number-frame"
  >
    140%
  </h3>
  <p
    className="fd-typography article-p-inline-number-frame"
    textStyle="article-p-inline-number-frame"
  >
    Curabitur quis luctus est, vitae sodales enim
  </p>
</div>
`;

exports[`NumberFrame should render align right correctly 1`] = `
<div
  className="inline-content cyferkader right"
>
  <h3
    className="fd-typography article-h-inline-number-frame"
    textStyle="article-h-inline-number-frame"
  >
    140%
  </h3>
  <p
    className="fd-typography article-p-inline-number-frame"
    textStyle="article-p-inline-number-frame"
  >
    Curabitur quis luctus est, vitae sodales enim
  </p>
</div>
`;

exports[`NumberFrame should render align right correctly 2`] = `
<div
  className="inline-content cyferkader right"
>
  <h3
    className="fd-typography article-h-inline-number-frame"
    textStyle="article-h-inline-number-frame"
  >
    140%
  </h3>
  <p
    className="fd-typography article-p-inline-number-frame"
    textStyle="article-p-inline-number-frame"
  >
    Curabitur quis luctus est, vitae sodales enim
  </p>
</div>
`;
